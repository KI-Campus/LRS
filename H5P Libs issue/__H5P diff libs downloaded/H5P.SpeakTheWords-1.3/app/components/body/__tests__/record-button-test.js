jest.unmock('../record-button');

import React from 'react';
import RecordButton from '../record-button';
import renderer from 'react-test-renderer';

describe('Record Button', () => {
  const eventStoreMock = {on: jest.fn(), trigger: jest.fn()};
  const speechEngineMock = jest.fn();
  const test = (action) => {
    const component = renderer.create(
      <RecordButton
        eventStore={eventStoreMock}
        speechEngine={speechEngineMock}
        l10n={{
          listeningLabel: 'listening',
          speakLabel: 'speak'
        }}
      />
    );

    let tree = component.toJSON();
    action(tree, component);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  };

  const mouseEvent = (button) => {
    return {
      nativeEvent: {
        which: button
      },
      preventDefault: jest.fn()
    }
  };

  it('should render initially', () => {
    test(() => {});
  });

  it('should start recording on left mouse down', () => {
    test((tree) => {
      tree.props.onMouseDown(mouseEvent(1));
    })
  });

  it('should stop recording when clicked again', () => {
    test((tree, component) => {
      component.getInstance().setState({
        'listening': true
      });
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      tree.props.onMouseDown(mouseEvent(1));
    })
  });

  it('should not start recording on right mouse down', () => {
    test(tree => {
      tree.props.onMouseDown(mouseEvent(0));
    });
  });

  const keyEvent = (keyCode) => {
    return {
      which: keyCode,
      preventDefault: jest.fn()
    }
  };

  it('should start recording when pressing space', () => {
    test(tree => {
      tree.props.onKeyDown(keyEvent(32));
    });
  });

  it('should start recording when pressing enter', () => {
    test(tree => {
      tree.props.onKeyDown(keyEvent(13));
    });
  });

  it('should not start recording when pressing "a"', () => {
    test(tree => {
      tree.props.onKeyDown(keyEvent(65));
    });
  });

  it('should be possible to disable', () => {
    test((tree, component) => {
      component.getInstance().setState({disabled: true});
    });
  });

  it('should be possible to hide', () => {
    test((tree, component) => {
      component.getInstance().setState({hidden: true});
    });
  })
});
