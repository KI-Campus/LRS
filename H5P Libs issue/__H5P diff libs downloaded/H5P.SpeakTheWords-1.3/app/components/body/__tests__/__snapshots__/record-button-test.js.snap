// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record Button should be possible to disable 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-disabled"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;

exports[`Record Button should be possible to hide 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-hidden"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;

exports[`Record Button should not start recording on right mouse down 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;

exports[`Record Button should not start recording when pressing "a" 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;

exports[`Record Button should render initially 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;

exports[`Record Button should start recording on left mouse down 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-listening"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  listening
</button>
`;

exports[`Record Button should start recording when pressing enter 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-listening"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  listening
</button>
`;

exports[`Record Button should start recording when pressing space 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-listening"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  listening
</button>
`;

exports[`Record Button should stop recording when clicked again 1`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button h5p-listening"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  listening
</button>
`;

exports[`Record Button should stop recording when clicked again 2`] = `
<button
  className="h5p-speak-the-words-record h5p-joubelui-button"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  type="button"
>
  speak
</button>
`;
